automation_tools/__init__.py

```diff
@@ -1763,22 +1774,25 @@ def create_openstack_instance(instance_name, image_name, flavor_name, ssh_key):
         network=network.id
     )
     # Assigning floating ip to instance
+    timeup = time.time()+(int(timeout)*60)
     while True:
         try:
+            if time.time() > timeup:
+                print('The timeout for assigning the floating IP has reached!')
+                sys.exit(1)
             instance.add_floating_ip(floating_ip)
+            print('SUCCESS!! The flaoting IP has been assigned to instance!')
```

Comment:
Noted only Floating IP. The instance name is not required note as we are not using it anywhere to test.